src/Reflection/ReflectionClass.php

```diff
@@ -358,16 +364,16 @@ public function getExtensionName(): string|null
     /** @return list<ReflectionMethod> */
     private function createMethodsFromTrait(ReflectionMethod $method): array
     {
-        $methodModifiers = $method->getModifiers();
-        $methodHash      = $this->methodHash($method->getImplementingClass()->getName(), $method->getName());
+        $methodModifiers      = $method->getModifiers();
+        $lowerCasedMethodHash = $this->lowerCasedMethodHash($method->getImplementingClass()->getName(), $method->getName());
 
-        if (array_key_exists($methodHash, $this->traitsData['modifiers'])) {
+        if (array_key_exists($lowerCasedMethodHash, $this->traitsData['modifiers'])) {
             // PhpParser modifiers are compatible with PHP reflection modifiers
-            if ($this->traitsData['modifiers'][$methodHash] & Modifiers::VISIBILITY_MASK) {
-                $methodModifiers = ($methodModifiers & ~ Modifiers::VISIBILITY_MASK) | $this->traitsData['modifiers'][$methodHash];
+            if ($this->traitsData['modifiers'][$lowerCasedMethodHash] & Modifiers::VISIBILITY_MASK) {
+                $methodModifiers = ($methodModifiers & ~ Modifiers::VISIBILITY_MASK) | $this->traitsData['modifiers'][$lowerCasedMethodHash];
             }
 
-            if ($this->traitsData['modifiers'][$methodHash] & Modifiers::FINAL) {
+            if ($this->traitsData['modifiers'][$lowerCasedMethodHash] & Modifiers::FINAL) {
                 $methodModifiers |= Modifiers::FINAL;
             }
```

Comment:
I think this calls for a small `@internal` class that keeps this trait data: the fact that we need to look up by lowercase hash seems to be something akin a `Map<TTraitMethod, Modifiers>`.